subsys/bluetooth/host/gatt.c

```diff
@@ -5140,6 +5153,57 @@ int bt_gatt_store_ccc(uint8_t id, const bt_addr_le_t *addr)
 	return 0;
 }
 
+uint8_t convert_to_public_on_match(const struct bt_gatt_attr *attr, uint16_t handle,
+			void *user_data)
+{
+	struct _bt_gatt_ccc     *ccc;
+	struct device_addresses *device_addresses   = user_data;
+
+	/* Check if attribute is a CCC */
+	if (attr->write != bt_gatt_attr_write_ccc) {
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	ccc = attr->user_data;
+
+	/* Copy the device's public address to the config's address if the config's address is the
+	 * same as the device's private address
+	 */
+	for (size_t i = 0; i < ARRAY_SIZE(ccc->cfg); i++) {
+		if (bt_addr_le_cmp((const bt_addr_le_t *)&ccc->cfg[i].peer,
+							device_addresses->private_addr) == 0) {
+			bt_addr_le_copy(&ccc->cfg[i].peer, device_addresses->public_addr);
+#ifndef CONFIG_BT_CTLR_ALLOW_SAME_PEER_CONN
+			break;
+#endif /* CONFIG_BT_CTLR_ALLOW_SAME_PEER_CONN */
+		}
+	}
+	return BT_GATT_ITER_STOP;
+}
+
+void bt_gatt_update_ccc_cfg_addr(const bt_addr_le_t *private_addr, const bt_addr_le_t *public_addr)
+{
+	struct device_addresses user_data = {.private_addr = private_addr,
+		.public_addr = public_addr};
```

Comment:
There should be a space before `}` and after `{`. You might in general make this more readable with something like:
```
struct device addresses addr = {
        .private_addr = private_addr,
        .public_addr = public_addr,
};
```